### PR TITLE
Enable T3K Resnet Tests

### PR DESCRIPTION
**Pull request by tt-aho**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/10244

### Problem description
Enable T3K Resnet functionality tests in CI now that maxpool changes have been merged

### What's changed
Enable T3K Resnet functionality tests in CI. This also includes a bug fix for reshard not handling optional output tensor properly which is needed to enable Resnet

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by TT-BrianLiu — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -66,14 +66,14 @@
...
+| [Falcon40B](./models/demos/t3000/falcon40b)        | [v0.51.0-rc13](https://github.com/tenstorrent/tt-metal/tree/v0.51.0-rc13) | Tensor Parallel    | 129th               |  32                   | 5.3 t/s/u - 168 t/s          |  12.2 t/s/u - 390 t/s        |   36 t/s/u      |
 | [Mixtral7Bx8](./models/demos/t3000/mixtral8x7b)    | [v0.51.0-rc13](https://github.com/tenstorrent/tt-metal/tree/v0.51.0-rc13) | Tensor Parallel    | 129th               |  32                   | 13.3 t/s/u - 426 t/s         |  21.4 t/s/u - 685 t/s        |   33 t/s/u      |
-| ResNet50                                           |                                                                           | Data Parallel      | coming soon         |                       |                              |                              |                 |
+| [ResNet-50](./models/demos/ttnn_resnet)            |                                                                           | Data Parallel      |                     |  128                  | 31,700                       |  44,400                      |   56,000        |
```
Comment (ttmchiou) [inline README.md:76]: Should we match units here?
All the other entries have t/s/u units or etc.

Comment (tt-aho) [reply README.md:76]: Units don't match. Those are generative models and reports 2 different units based on tokens.
Resnet units are  frames/images.

This is also how the single device tables are set up.

image

**Review by ttmchiou — APPROVED** [review]

**Comment by ttmchiou** [comment]
make sure https://github.com/tenstorrent/tt-metal/actions/runs/10222401824 passes before merging

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by tt-aho** [comment]
Passing CI:
Post Commit: https://github.com/tenstorrent/tt-metal/actions/runs/10243410057
Nightly: https://github.com/tenstorrent/tt-metal/actions/runs/10244206423
Model Perf: https://github.com/tenstorrent/tt-metal/actions/runs/10243414623
T3K Frequent: https://github.com/tenstorrent/tt-metal/actions/runs/10243411546
T3K Perf: https://github.com/tenstorrent/tt-metal/actions/runs/10243410771

**Pull request merged**